crates/pixi_build_frontend/src/protocols/builders/pixi.rs

```diff
@@ -195,17 +196,20 @@ impl ProtocolBuilder {
 
         // If we get here the tool is not overridden, so we use the isolated variant
         let build_system = &self.package_manifest.build;
-        let specs = [(&build_system.backend.name, &build_system.backend.spec)]
+        let mut specs = [(&build_system.backend.name, &build_system.backend.spec)]
             .into_iter()
             .chain(build_system.additional_dependencies.iter())
             .map(|(name, spec)| {
                 spec.clone()
                     .try_into_nameless_match_spec(channel_config)
                     .map(|spec| MatchSpec::from_nameless(spec, Some(name.clone())))
             })
-            .collect::<Result<_, _>>()
+            .collect::<Result<Vec<_>, _>>()
             .map_err(FinishError::SpecConversionError)?;
 
+        // Ensure that backend uses the same interface as Pixi
+        specs.push(PIXI_BUILD_INTERFACE_MATCHSPEC.clone());
```

Comment:
Instead of spec we should add a constraint and then check if the package is part of the solution. Only then are we certain that a package in the solution depends on it and it was not pixi that introduced the package.